test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -506,7 +507,7 @@ func (rc *ResourceConsumer) WaitForReplicas(ctx context.Context, desiredReplicas
 // EnsureDesiredReplicasInRange ensure the replicas is in a desired range
 func (rc *ResourceConsumer) EnsureDesiredReplicasInRange(ctx context.Context, minDesiredReplicas, maxDesiredReplicas int, duration time.Duration, hpaName string) {
 	interval := 10 * time.Second
-	err := wait.PollUntilContextTimeout(ctx, interval, duration, true, func(ctx context.Context) (bool, error) {
+	gomega.Consistently(ctx, framework.HandleRetry(func(ctx context.Context) (bool, error) {
```

Comment:
I´m not sure if this is dealing with the specific error from the bug you linked , @pohly knows better these functions

Reply:
followed https://github.com/kubernetes/kubernetes/pull/121583 as an example, which was fixing a similar error message. I wasn't sure I needed the `HandleRetry` here though

Reply:
tracking feedback in https://github.com/kubernetes/kubernetes/pull/121861/files#r1392089576